kernel/reduction.ml

```diff
@@ -488,22 +488,22 @@ and eqappr cv_pb l2r infos (lft1,st1) (lft2,st2) cuniv =
            we throw them away *)
         if not (is_empty_stack v1 && is_empty_stack v2) then
           anomaly (Pp.str "conversion was given ill-typed terms (FLambda).");
-        let (x1,ty1,bd1) = destFLambda mk_clos hd1 in
-        let (_,ty2,bd2) = destFLambda mk_clos hd2 in
+        (* We assume terms well-typed in the same supertype, so we don't compare the domains *)
+        let (x1,_ty1,bd1) = destFLambda mk_clos hd1 in
+        let (_,_ty2,bd2) = destFLambda mk_clos hd2 in
```

Comment:
Sadly in the past this killed performance https://github.com/coq/coq/wiki/Performance-experiments#skipping-conversion-of-parameters-of-constructors-and-of-the-domain-of-lambdas-failed

Reply:
Ah, ok. So this means that we may require a light conversion (where all sorts are identified)?

Reply:
Maybe.
I guess that would look like 547448a584c06c682e317103ac6f5bf4cf05f75d (pushed on top of this PR)